packages/cdk8s-plus/src/deployment.ts

```diff
@@ -186,27 +149,42 @@ export class DeploymentSpecDefinition {
   }
 
   /**
-   * The labels this deployment will match against in order to select pods.
+   * Expose a deployment via a service.
    *
-   * Returns a a copy. Use `selectByLabel()` to add labels.
+   * This is equivalent to running `kubectl expose deployment <deployment-name>`.
+   *
+   * @param port The port number the service will bind to.
+   * @param options Options.
    */
-  public get labelSelector(): Record<string, string> {
-    return { ...this._labelSelector };
+  public expose(port: number, options: ExposeOptions = {}): Service {
+    const service = new Service(this, 'Service', {
+      type: options.serviceType ?? ServiceType.CLUSTER_IP,
+    });
+
+    service.addDeployment(this, port);
+    return service;
+  }
+
+  public addContainer(container: Container): void {
+    return this._podTemplate.addContainer(container);
+  }
+
+  public addVolume(volume: Volume): void {
+    return this._podTemplate.addVolume(volume);
   }
-  
+
+
   /**
    * @internal
    */
   public _toKube(): k8s.DeploymentSpec {
     return {
       replicas: this.replicas,
-      template: {
-        metadata: this.podMetadataTemplate.toJson(),
-        spec: this.podSpecTemplate._toKube(),
-      },
+      template: this._podTemplate._toPodTemplateSpec(),
```

Comment:
toKube maybe?

Reply:
I can't :\ 

`_toKube` is already implemented with a different return value in `PodSpec`. See https://github.com/awslabs/cdk8s/pull/347#discussion_r507086143

Reply:
oh you mean just without the underscore?